analyses/count-contributions/02-format-contributions.Rmd

```diff
@@ -0,0 +1,182 @@
+
+```{r}
+name_recoding <- c(
+  jashapiro = "Joshua Shapiro",
+  Stephanie = "Stephanie J. Spielman",
+  Bailey = "Bailey Farrow",
+  runjin326 = "Run Jin",
+  Mateusz = "Mateusz P. Koptyra",
+  tkoganti = "Teja Koganti",
+  NNoureen = "Nighat Noureen",
+  `Jo Lynne` = "Jo Lynne Rokita",
+  `e-t-k` = "Ellen Kephardt",
+  yangyangclover = "Yang Yang",
+  PichaiRaman = "Pichai Raman",
+  hbeale = "Holly Beale"
+)
+```
+
+## Module contributions
+
+There are some modules that did not generate results that are included in the manuscript.
+For our purposes, we want to skip those lists of contributors.
+The not included list is shorter than the included list, so that's why we'll take this approach!
+
+```{r}
+# We're not counting these because they didn't make it to the manuscript
+not_included_analyses <- c(
+  "cnv-comparison",
+  "comparative-RNASeq-analysis",
+  "compare-gistic",
+  "count-contributions",
+  "create-subset-files",
+  "molecular-subtyping-ATRT",
+  "molecular-subtyping-SHH-tp53",
+  "mutect2-vs-strelka2",
+  "rna-seq-composition",
+  "selection-strategy-comparison",
+  "sex-prediction-from-RNASeq",
+  "ssgsea-hallmark",
+  "tcga-capture-kit-investigation",
+  "sample-distribution-analysis"
+)
+```
+
+All of the _relevant files_ are those TXT files in `components/` that do not match one of the not included modules listed above.
+We're also going to skip `components/total_contributions.txt` for now. 
+
+```{r}
+# All of the files that were included in components directory
+contribution_files <- list.files(components_dir)
+
+# We'll count the files that come from modules that are included in the 
+# manuscript by removing the modules that *are not* included
+count_files <- contribution_files[str_detect(contribution_files, 
+                                             pattern = paste(
+                                               not_included_analyses, 
+                                               collapse = "|"),
+                                             negate = TRUE)]
+
+# We're also not going to look at total contributions here
+count_files <- setdiff(count_files, "total_contributions.tsv")
+
+count_files
+```
+
+Now read in the relevant contribution files to a data frame that we can use to count.
+
+```{r}
+count_contributions_df <- suppressMessages(  # This is chatty!
+  # For all the files we'll consider
+  file.path(components_dir, count_files) %>%
+    # Read them in, but skip the first (blank) line and don't use colnames
+    # This step produces a list
+    map(read_tsv, skip = 1, col_names = FALSE) %>%
+    # Use the file names themselves to set the names of the list
+    # This will be the name of the module
+    set_names(str_remove(count_files, "_contributors.txt")) %>% 
+    # Create a data frame where we use the module names to keep track of where
+    # the data frame of contributors came from
+    bind_rows(.id = "module") %>% 
+    distinct()
```

Comment:
Setting author column name inline.
```suggestion
    # Read them in, but skip the first (blank) line and set the (only) column name
    # This step produces a list
    map(read_tsv, skip = 1, col_names = "author") %>%
    # Use the file names themselves to set the names of the list
    # This will be the name of the module
    set_names(str_remove(count_files, "_contributors.txt")) %>% 
    # Create a data frame where we use the module names to keep track of where
    # the data frame of contributors came from
    bind_rows(.id = "module") %>% 
    distinct()
```

Reply:
I will add all these suggestions in one commit and test in the `ccdlopenpbta/open-pbta` container to see how it goes. If anything goes sideways, I am going to revert them all in one foul swoop since I know what we currently have wokrs!